specification/assets/gherkin/metadata.feature

```diff
@@ -0,0 +1,25 @@
+@Metadata
+Feature: Metadata
+
+  Background:
+    Given a provider is registered
+
+  Scenario: Returns metadata
+    Given a Boolean-flag with key "metadata-flag" and a default value "true"
+    When the flag was evaluated with details
+    Then the resolved metadata value "string" with type "String" should be "1.0.2"
+    And the resolved metadata value "integer" with type "Integer" should be "2"
+    And the resolved metadata value "double" with type "Double" should be "0.1"
+    And the resolved metadata value "boolean" with type "Boolean" should be "true"
```

Comment:
We could probably use a table for this, with columns for the metadata key, type, and value.